File: atcoder/beginner_contest/145_20191116/d.cpp
#include <iostream>
typedef long long int ll;


using namespace std;

ll MOD = 1000000000LL + 7LL;


ll pow_mod(ll a, ll n) {
    if (n == 0LL) {
        return 1LL;
    }
    ll p = pow_mod(a, n / 2);
    ll temp = (p * p) % MOD;
    if (n % 2 == 0LL) {
        return temp;
    } else {
        return (a * temp) % MOD;
    }
}

int main(int argc, char const *argv[]) {
    ll x, y;
    cin >> x >> y;
    if ((x + y) % 3 != 0) {
        cout << 0 << endl;
        return 0;
    }
    ll n = (x + y) / 3;
    x -= n;
    y -= n;

    if (x < 0LL or y < 0LL) {
        cout << 0 << endl;
        return 0;
    }
    ll ans = 1LL;
    for (ll i = x + 1; i < x + y + 1; i++) {
        ans = (ans * i) % MOD;
    }
    for (ll i = 1LL; i < y + 1; i++) {
        ans = (ans * pow_mod(i, MOD - 2)) % MOD;
    }
    cout << ans << endl;
    return 0;
}

File: atcoder/beginner_contest/145_20191116/d.py
MOD = 10 ** 9 + 7


def pow_mod(a, n):
    if n == 0:
        return 1
    p = pow_mod(a, n // 2)
    if n % 2 == 0:
        return (p * p) % MOD
    else:
        return (a * p * p) % MOD
        

def main():
    x, y = map(int, input().split())
    if (x + y) % 3 != 0:
        return 0
    
    n = (x + y) // 3
    x -= n
    y -= n
    if x < 0 or y < 0:
        return 0
    else:
        ans = 1
        for i in range(x + 1, x + y + 1):
            ans = (ans * i) % MOD
        for i in range(1, y + 1):
            ans = (ans * pow_mod(i, MOD - 2)) % MOD
        return ans


if __name__ == "__main__":
    print(main())